### PR TITLE
Refactor package structure and rename types & fields

### LINKED ISSUE #15: Split code and logic into separate packages
Currently, it's hard to separate _anything_, because of cyclical imports. So, I need to separate out the current types and functions in order to allow smaller package in new code we write.

### PR DESCRIPTION
**Pull request by kroppt**
This precedes a big menu change, hence why I avoided changing those type names. Other than that, feel free to point out any and all concerns.

I require both of you to review it before I merge it.

### CODE REVIEW

File: pkg/ui/ui.go [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+	if alpha, ok := mask.(*image.Alpha); ok {
+		diff := image.Point{rec.Dx(), rec.Dy()}
+		tofile := alpha.SubImage(image.Rectangle{maskp, maskp.Add(diff)})
+		if f, err := os.OpenFile(fileName, os.O_RDWR|os.O_CREATE, 0755); err != nil {
+			png.Encode(f, tofile)
+			return err
+		}
+	}
+	return nil
+}
```
Comment (gregjohnson2017) [inline pkg/ui/ui.go]: I think this should be in font/font.go (even tho it's unused, might as well keep it around and put it in the right place)

File: pkg/ui/ui.go [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+	AlignCenter
+	// AlignRight puts the right side at the x coordinate
+	AlignRight
+)
+
+// Align holds vertical and horizontal alignments
+type Align struct {
+	V AlignV
+	H AlignH
+}
```
Comment (gregjohnson2017) [inline pkg/ui/ui.go:52]: Should this be in font/font.go? I'm not sure

File: pkg/ui/ui.go [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+		gl.GetProgramiv(prog, gl.INFO_LOG_LENGTH, &logLength)
+
+		log := strings.Repeat("\x00", int(logLength+1))
+		gl.GetProgramInfoLog(prog, logLength, nil, gl.Str(log))
+
+		return 0, fmt.Errorf("failed to compile program: %v", log)
+	}
+
+	return prog, nil
+}
```
Comment (gregjohnson2017) [inline pkg/ui/ui.go]: Does this OpenGL stuff belong in its own location?

File: pkg/image/view.go [hunk 4]
```diff
@@ -386,10 +384,30 @@ func (iv *ImageView) OnResize(x, y int32) {
...
+	if surf.Format.Format != uint32(sdl.PIXELFORMAT_RGBA32) {
+		convertedSurf, err := surf.ConvertFormat(uint32(sdl.PIXELFORMAT_RGBA32), 0)
+		surf.Free()
+		if err != nil {
+			return nil, err
+		}
+		return convertedSurf, nil
+	}
+
+	return surf, err
```
Comment (gregjohnson2017) [inline pkg/image/view.go]: I agree that it makes more sense to put this here instead of ui.go

**Review by gregjohnson2017 — CHANGES_REQUESTED** [review]

File: pkg/font/font.go [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+	return buffer
+}
+
+type fontKey struct {
+	fontName string
+	fontSize int32
+}
+
+type Info struct {
+	TextureID  uint32     // OpenGL texture ID of cached glyph data
```
Comment (kropptrevor) [inline pkg/font/font.go]: It seems like this should not be capitalized

**Review by kropptrevor — CHANGES_REQUESTED** [review]

File: pkg/ui/ui.go (hunk 1 above)
Comment (kroppt) [reply pkg/ui/ui.go]: Done.

File: pkg/ui/ui.go (hunk 2 above)
Comment (kroppt) [reply pkg/ui/ui.go:52]: I moved all the graphics stuff to its own package, so the purpose of this should be more clear. I expect this to be used in the future by non-font UI stuff.

File: pkg/ui/ui.go (hunk 3 above)
Comment (kroppt) [reply pkg/ui/ui.go]: Done.

File: pkg/font/font.go (hunk 5 above)
Comment (kroppt) [reply pkg/font/font.go]: Done.

**Review by gregjohnson2017 — APPROVED** [review]

**Review by gregjohnson2017 — APPROVED** [review]

**Review by kropptrevor — APPROVED** [review]

**Pull request merged**